File: app/src/main/java/com/sunexample/demoforandroidxandkotlin/AppContext.java
package com.sunexample.demoforandroidxandkotlin;

import android.app.Application;
import android.content.Context;


public class AppContext extends Application {
    private static final String TAG = "AppContext";
    public static Context mContext;


    @Override
    public void onCreate() {
        super.onCreate();
        mContext = this;




      /*  if (LeakCanary.isInAnalyzerProcess(this)) {
            // This process is dedicated to LeakCanary for heap analysis.
            // You should not init your app in this process.
            return;
        }
        LeakCanary.install(this);
        // Normal app init code...*/

    }

    public static Context getAppContext() {
        return mContext;
    }
}

File: app/src/main/java/com/sunexample/demoforandroidxandkotlin/AppContext.kt
package com.sunexample.demoforandroidxandkotlin

import android.app.Application
import android.content.Context
import org.litepal.LitePal

class AppContext : Application() {
    override fun onCreate() {
        super.onCreate()
        appContext = this

        LitePal.initialize(this)
    }

    companion object {
        private const val TAG = "AppContext"
        var appContext: Context? = null
    }
}